### DISCUSSION TITLE
Error in pushing PA to github

### DISCUSSION THREAD

**Discussion by MeteorTony** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#68] [Q&A] [answered]
Encountered some problems when using "git push -u origin main".
<img width="586" alt="Screenshot (7)" src="https://user-images.githubusercontent.com/61377153/135754116-e03de67c-abe7-44c3-b166-6ebd6f38bc90.png">
should I use "push -f origin main" to fix it? else what should I do?

**Comment by Derppening** [comment, marked as answer]
You committed two profiling files into your Git repository, which are so big that GitHub rejected your commit.

You should remove these files from your repository, and commit the changes **with the "Amend" checkbox ticked**. This should make the two large files disappear without creating a new commit.

**Reply by MeteorTony** [reply]
where can I find the "Amend" checkbox? Since I used to type in commands to deal with git, now I have a problem in finding the checkbox....

**Reply by Derppening** [reply]
- Delete the .hprof files
- Stage all changes (`git add .`)
- `git commit --amend`
- You probably don't need to change the commit message, so just save the file

Then you can re-push the changes. If that doesn't work, there is still a way to fix it, but it will be too complex to describe here.

**Reply by MeteorTony** [reply]
thanks, I guess the problem is already fixed!
I delete the .hprof files and use "git reset --soft Head ~1" to delete the last commit and commit something new, it seems OK now.

**Reply by MeteorTony** [reply]
There's still a problem, the finalized zipped file is around 300 MB which I can't upload to CASS(I don't know why....)...

**Reply by MeteorTony** [reply]
My original file is like 900 KB only

**Reply by MeteorTony** [reply]
Can I re-download it from my Github repository to submit?

**Reply by Derppening** [reply]
Run the "Clean" configuration in IntelliJ and zip it again.

If the file is still too big, you may just download the ZIP from GitHub and upload it. Just be reminded that the ZIP should have a `github-url.txt` file with a link to your Github repository.
